### DISCUSSION TITLE
How to render Quarto metadata in .tex partials ?

### DISCUSSION THREAD

**Discussion by martin-g** [quarto-dev/quarto-cli#7692] [Q&A] [answered]
### Description

Hi!

I have a custom metadata in _quarto.yml:
```yaml
...
mycustom: value
...
```

I can render it in .qmd files with: `{{< meta mycustom >}}` ! So far so good!

But I also need to overwrite the `title.tex` template with a custom one. 
What is the way/syntax to render the value of `mycustom` metadata ?

Thanks!
Martin

**Comment by mcanouil** [comment]
Did you look at the documentation on the matter?
https://quarto.org/docs/journals/templates.html

**Reply by martin-g** [reply]
Thanks for the quick answer, @mcanouil !

I did look at that page but using `$mycustom$` didn't work. The referenced page shows only how to render Pandoc specific [metadata](https://pandoc.org/chunkedhtml-demo/6.2-variables.html) like `author`, `title` and `date` but does not say anything about custom Quarto metadata.

Here is the line in my title.tex that tries to render it:
```
...
\noindent {\Large \color{logodarkgrey} \fontfamily{phv}\selectfont $mycustom$ } \\ 
...
```

but instead of `value` it renders `mycustom`

**Reply by cderv** [reply]
If I try a simple example like this 

````
---
title: "Quarto Playground"
format: 
  latex:
    template-partials: 
      - title.tex
mycustom: value
---

COntent

````

with `title.tex` being 
````
\noindent {\Large \color{logodarkgrey} \fontfamily{phv}\selectfont $mycustom$ } \\
````

Then in the rendered .tex file I get 

````
\hypersetup{
  pdftitle={Quarto Playground},
  colorlinks=true,
  linkcolor={blue},
  filecolor={Maroon},
  citecolor={Blue},
  urlcolor={Blue},
  pdfcreator={LaTeX via pandoc}}

\noindent {\Large \color{logodarkgrey} \fontfamily{phv}\selectfont value } \\
\begin{document}
\maketitle
COntent



\end{document}
````

See that `value` is there. 

Also this example is a dummy one (no rendering probably to PDF because of error) - it is just to show the value is passed.

**Reply by martin-g** [reply]
@cderv Thanks! What command did you use to render the `.tex` ? If I use `--to tex` then Quarto fails with
```
ERROR: Unknown format tex

Error: Unknown format tex
    at resolveFormats (file:///opt/quarto/bin/quarto.js:83183:19)
...
```

My next guess is that you actually tried to render PDF (i.e. `--to pdf`) and told Quarto to preserve the intermediate `.tex` file (`keep-tex`) but you say `no rendering probably to PDF because of error`, so it sounds like you didn't try to render a PDF...

I am not sure what is wrong with my setup but I see `mycustom` in the intermediate `.tex` file too.

My Quarto version is 1.3.450.

**Reply by martin-g** [reply]
I think I found the problem!

My _quarto.yml contains:
```
...
format:
  pdf:
    ...
    template-partials:
      - partials/before-body.tex
     ...
```

and `partials/before-body.tex` is:
```
MyCustom: $mycustom$

% Title Page --------------------------------------------------------
\input{common/title}
% Disclaimer
\input{common/legal} 

\newpage
\tableofcontents
\newpage
```

`$mycustom$` in before-body.tex is replaced with `value` ! This works!

But I actually need to use Quarto metadata in `common/title.tex` which is included in `before-body.tex` and there `$mycustom$` is not interpolated to `value`!

For now my workaround is to inline `title.tex` into `before-body.tex`.

**Reply by cderv** [reply, marked as answer]
> What command did you use to render the .tex ? If I use --to tex then Quarto fails with

This can be found in the `format` value. `format: latex` which leads to `--to latex` at command line. `--to tex` is not valid. 

> But I actually need to use Quarto metadata in common/title.tex which is included in before-body.tex and there $mycustom$ is not interpolated to value!

I don't understand exactly what you want to do. You may need to learn more Pandoc's Template, and how Partials works (https://pandoc.org/MANUAL.html#partials)

* What is `common/title.tex` ? If this is not a Pandoc's template, no templating expansion will happens. And I see  `\input{common/title}` which is not a template call, but a LaTeX include of a an external tex file. Pandoc won't expand anything in it by default. 

> For now my workaround is to inline title.tex into before-body.tex.

This seems like a good way - or you may need to consider `common/title.tex` also as a Pandoc partials that should be included and expanded by Pandoc. 

Overall, your question is about how Pandoc's templating is working. You should try different things to understand how that works

**Reply by martin-g** [reply]
Thank you for the clarifications, @cderv !

**Reply by TomBener** [reply]
I encountered the same problem. The value of custom metadata in `before-body.tex` was expanded, but the same content in `title.tex` didn't.

**Reply by cderv** [reply]
@TomBener did the above clarified it for you ? 

If not, please open a new discussion as this one is a answered by the OP - or clarify what you are looking for in another post below. 

We can just mark only one post as an answer in a discussion - so always ok to open a new discussion (possibly linking to this one)
